### DISCUSSION TITLE
How to plot the generated volume ?

### DISCUSSION THREAD

**Discussion by jeremieglt** [IMSY-DKFZ/simpa#393] [Q&A]
Hi,

I want to use SIMPA to generate images with customed (and non random) tissue volumes for my master's thesis and I am still getting a first hands on right now. I had a little interrogation about volume generation : is there a simple way to visualize in 3D the tissue volumes that we generate at the very beggining of the pipeline ? I didn't see any function or example going in that way and I tried to just take the 3D array of "Tags.DATA_FIELD_SEGMENTATION" and plot it with with different methods (pyplot, plotly and blender) but nothing gave any good result. I however saw in your paper that you had some nice visualizations : did you use an external method or did I miss something in the package ?

Thank you in advance for your reading,
Jeremie

**Comment by kdreher** [comment]
Hi Jeremi,
I would say in most cases, it is not really needed to do a full 3D visualization. Usually, just plotting different viewpoints along different axes is sufficient to find out what's going on. These are for example implemented in our [minimal optical example](https://github.com/IMSY-DKFZ/simpa/blob/main/simpa_examples/minimal_optical_simulation.py).

If you really want to have something like in the simpa paper, you can reproduce the results in the [paper repository](https://github.com/IMSY-DKFZ/simpa_paper_experiments). Especially the [tissue generation folder](https://github.com/IMSY-DKFZ/simpa_paper_experiments/tree/master/experiments/tissue_generation) would be interesting for that. Note, that 3D voxelized visualizations might take a while because matplotlib plots each voxel in a volume as a separate vector element.

**Reply by jeremieglt** [reply]
Hi Kris,
Thanks a lot for your answer and help. With the current volume that I want to simulate, only plotting 2D slices could indeed be a good first approximation but 3D plotting could help me at some point to verify that elements are correctly placed in space.

**Reply by kdreher** [reply]
Ok, then I guess the way we visualize segmentation masks in the tissue generation folder of the paper repository should be helpful.
